index.js

```diff
@@ -84,8 +84,8 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
-  var opt = options || {};
-  var enc = opt.encode || encode;
+  var opt = typeof options === 'object' ? options : {};
+  var enc = typeof opt.encode === 'function' ? opt.encode : encode;
```

Comment:
Actually, it didn't always error before. Any falsy value for `opt.encode` would fall through to the default `encode` function. Maybe the intent of the original logical OR was to see if the "encode" property was defined on the options object or not, if that's the case, what about using `typeof opt.encode !== 'undefined' ? opt.encode : encode`. That should allow any defined value for `opt.encode` which isn't a function to error out.